libqtile/core/manager.py

```diff
@@ -1219,89 +1223,32 @@ def spawn(self, cmd: str | list[str], shell: bool = False) -> int:
             args = shlex.split(cmd)
         else:
             args = list(cmd)
-            cmd = subprocess.list2cmdline(args)
 
         to_lookup = args[0]
-        if shell:
-            args = ["/bin/sh", "-c", cmd]
 
         if shutil.which(to_lookup) is None:
             logger.error("couldn't find `%s`", to_lookup)
             return -1
 
-        r, w = os.pipe()
-        pid = os.fork()
-        if pid < 0:
-            os.close(r)
-            os.close(w)
-            return pid
-
-        if pid == 0:
-            os.close(r)
-
-            # close qtile's stdin, stdout, stderr so the called process doesn't
-            # pollute our xsession-errors.
-            os.close(0)
-            os.close(1)
-            os.close(2)
-
-            pid2 = os.fork()
-            if pid2 == 0:
-                os.close(w)
-                try:
-                    # if qtile was installed in a virutal env, we don't
-                    # necessarily want to propagate that to children
-                    # applications, since it may change e.g. the behavior
-                    # of shells that spawn python applications
-                    del os.environ["VIRTUAL_ENV"]
-                except KeyError:
-                    pass
-
-                # Open /dev/null as stdin, stdout, stderr
-                try:
-                    fd = os.open(os.devnull, os.O_RDWR)
-                except OSError:
-                    # This shouldn't happen, catch it just in case
-                    pass
-                else:
-                    # For Python >=3.4, need to set file descriptor to inheritable
-                    try:
-                        os.set_inheritable(fd, True)
-                    except AttributeError:
-                        pass
+        env = os.environ.copy()
+        try:
+            env.pop("VIRTUAL_ENV")
+        except KeyError:
+            pass
 
-                    # Again, this shouldn't happen, but we should just check
-                    if fd > 0:
-                        os.dup2(fd, 0)
+        proc = subprocess.Popen(args, shell=shell, env=env)
 
-                    os.dup2(fd, 1)
-                    os.dup2(fd, 2)
+        if not self._spawned_processes:
+            self._zombie_killer = self.call_later(10, self._kill_zombies)
 
-                try:
-                    os.execvp(args[0], args)
-                except OSError:
-                    # can't log here since we forked :(
-                    pass
+        self._spawned_processes.append(proc)
```

Comment:
this would leak `proc` if called in parallel with _kill_zombies

what is qtile threading policy, can it be relied that Qtile methods are only called from a single thread?